zero/src/rpc/jerigon.rs

```diff
@@ -53,9 +121,76 @@ where
                 compact: hex::decode(block_witness.strip_prefix("0x").unwrap_or(&block_witness))
                     .context("invalid hex returned from call to eth_getWitness")?,
             }),
-            txn_info: tx_results.into_iter().map(|it| it.result).collect(),
+            txn_info,
             code_db: Default::default(),
         },
         other_data,
     })
 }
+
+/// Processes the transactions in the given block and updates the code db.
+pub async fn process_transactions<'i, I, ProviderT, TransportT>(
+    block: &Block,
+    provider: &ProviderT,
+    tx_traces: I,
+) -> anyhow::Result<Vec<Option<JumpDestTableWitness>>>
+where
+    ProviderT: Provider<TransportT>,
+    TransportT: Transport + Clone,
+    I: Iterator<Item = &'i BTreeMap<H160, TxnTrace>>,
+{
+    let futures = block
+        .transactions
+        .as_transactions()
+        .context("No transactions in block")?
+        .iter()
+        .zip(tx_traces)
+        .map(|(tx, tx_trace)| process_transaction(provider, tx, tx_trace))
+        .collect::<FuturesOrdered<_>>();
+
+    let vec_of_res = futures.collect::<Vec<_>>().await;
+    vec_of_res.into_iter().collect::<Result<Vec<_>, _>>()
```

Comment:
```suggestion
    futures
        .collect::<Vec<_>>()
        .await
        .into_iter()
        .collect::<Result<Vec<_>, _>>()
```

Reply:
Btw. this function uses `FuturesOrdered` that was new to me. But do we really care about the order in which these futures are resolved?  Should I prefer `FuturesUnordered` here?

Reply:
`FuturesUnordered` should also be fine. I don't think there is a big difference.